clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitRequest.java

```diff
@@ -101,18 +103,18 @@ public static List<OffsetCommitResponseTopic> getErrorResponseTopics(
                                            .setPartitionIndex(requestPartition.partitionIndex())
                                            .setErrorCode(e.code()));
             }
-            responseTopicData.add(new OffsetCommitResponseTopic()
-                    .setName(entry.name())
-                    .setPartitions(responsePartitions)
-            );
+            OffsetCommitResponseTopic responseTopic = new OffsetCommitResponseTopic()
+                .setTopicId(version >= 9 ? entry.topicId() : Uuid.ZERO_UUID)
+                .setName(version < 9 ? entry.name() : null);
```

Comment:
I think that we could just set both the name and the id all the time as the fields are ignorable. The serialization framework will do the right thing based on the version. We could also remove `version` from the arguments.